python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,506 @@
+#!/usr/bin/python
+import logging
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from astropy.time import Time
+from lsst.summit.utils.efdUtils import getEfdData
+from lsst.summit.utils.tmaUtils import TMAEvent, TMAEventMaker
+from plots import inertia_compensation_system
+
+# TMAEventMaker needs to be instantiated only once.
+event_maker = TMAEventMaker()
+
+__all__ = ["M1M3ICSAnalysis"]
+
+
+class M1M3ICSAnalysis:
+    """
+    Evaluate the M1M3 Inertia Compensation System's performance by
+    calculating the minima, maxima and peak-to-peak values during a
+    slew. In addition, calculates the mean, median and standard deviation
+    when the slew has contant velocity or zero acceleration.
+
+    Parameters
+    ----------
+    event : `lsst.summit.utils.tmaUtils.TMAEvent`
+        Abtract representation of a slew event.
+    """
+
+    def __init__(
+        self,
+        event: TMAEvent,
+        inner_pad: float = 0.0,
+        outer_pad: float = 0.0,
+        n_sigma: float = 1.0,
+        logger: logging.Logger | None = None,
+    ):
+        if logger is None:
+            self._make_logger()
+        else:
+            self.logger = logger
```

Comment:
consider renaming `self.logger` to `self.log`. Also, instead of just assigning `self.logger = logger` consider making it a child:

```
self.logger = logger.getChild(type(self).__name__)
```

Also, I would suggest you remove `_make_logger` and simplify this by doing something like:

```
self.log = logger.getChild(type(self).__name__) if logger is not None else logging.getLogger(type(self).__name__)
```